Gems/Atom/RHI/Vulkan/Code/Source/RHI/CommandList.cpp

```diff
@@ -688,8 +689,8 @@ namespace AZ
             if (interval != InvalidInterval)
             {
                 uint32_t numBuffers = interval.m_max - interval.m_min + 1;
-                AZStd::vector<VkBuffer> nativeBuffers(numBuffers, VK_NULL_HANDLE);
-                AZStd::vector<VkDeviceSize> offsets(numBuffers, 0);
+                AZStd::fixed_vector<VkBuffer, RHI::Limits::Pipeline::StreamCountMax> nativeBuffers(numBuffers, VK_NULL_HANDLE);
+                AZStd::fixed_vector<VkDeviceSize, RHI::Limits::Pipeline::StreamCountMax> offsets(numBuffers, 0);
```

Comment:
I'm not sure if you'd want to add it to this PR, but there's another small gain to be had at this point:
```cpp
for (uint32_t i = 0; i < numBuffers; ++i)
{
    const RHI::StreamBufferView& bufferView = streams[i + interval.m_min];
    if (!bufferView.GetBuffer())
    {
        continue; // in this case entries in nativeBuffers and offsets are unchanged from their state pre-initialized above
    } 
    const auto* bufferMemoryView = static_cast<const Buffer*>(bufferView.GetBuffer())->GetBufferMemoryView();
    nativeBuffers[i] = bufferMemoryView->GetNativeBuffer();
    offsets[i] = bufferMemoryView->GetOffset() + bufferView.GetByteOffset();
}
```